File: android/app/src/main/java/com/dyjung/logindemo/domain/model/AuthProvider.kt
package com.dyjung.logindemo.domain.model

/**
 * 인증 제공자 타입
 */
enum class AuthProvider(val value: String) {
    /** 이메일 로그인 */
    EMAIL("email"),
    /** 카카오 로그인 */
    KAKAO("kakao"),
    /** 네이버 로그인 */
    NAVER("naver"),
    /** Apple 로그인 */
    APPLE("apple");

    /** 표시용 이름 */
    val displayName: String
        get() = when (this) {
            EMAIL -> "이메일"
            KAKAO -> "카카오"
            NAVER -> "네이버"
            APPLE -> "Apple"
        }

    companion object {
        fun fromValue(value: String): AuthProvider? =
            entries.find { it.value == value }
    }
}

File: ios/LoginDemo/Domain/Entities/AuthProvider.swift
//
//  AuthProvider.swift
//  LoginDemo
//
//  인증 제공자 열거형
//

import Foundation

/// 인증 제공자 타입
enum AuthProvider: String, Codable, Sendable, CaseIterable {
    /// 이메일 로그인
    case email = "EMAIL"

    /// 카카오 로그인
    case kakao = "KAKAO"

    /// 네이버 로그인
    case naver = "NAVER"

    /// Apple 로그인
    case apple = "APPLE"
    
    /// Google 로그인
    case google = "GOOGLE"

    /// 표시용 이름
    var displayName: String {
        switch self {
        case .email:
            return "이메일"
        case .kakao:
            return "카카오"
        case .naver:
            return "네이버"
        case .apple:
            return "Apple"
        case .google:
            return "Google"
        }
    }
}
